controller/eks-cluster-config-handler.go

```diff
@@ -619,17 +619,15 @@ func (h *Handler) generateAndSetNetworking(svc *cloudformation.CloudFormation, c
 		}
 
 		virtualNetworkString := getParameterValueFromOutput("VpcId", stack.Stacks[0].Outputs)
-		securityGroupsString := getParameterValueFromOutput("SecurityGroups", stack.Stacks[0].Outputs)
 		subnetIdsString := getParameterValueFromOutput("SubnetIds", stack.Stacks[0].Outputs)
 
-		if securityGroupsString == "" || subnetIdsString == "" {
-			return config, fmt.Errorf("no security groups or subnet ids were returned")
+		if subnetIdsString == "" {
+			return config, fmt.Errorf("no subnet ids were returned")
 		}
 
 		config = config.DeepCopy()
 		// copy generated field to status
 		config.Status.VirtualNetwork = virtualNetworkString
-		config.Status.SecurityGroups = strings.Split(securityGroupsString, ",")
```

Comment:
We want to decouple this but not remove it, so this can be removed from this block and the one above it and then set on the status if it is not empty.

Reply:
Security groups need to be created with a VPC ID. Therefore, if the user does not provide a customer VPC and Rancher creates one for them, then they cannot provide any security groups. So the security groups will always be empty if Rancher creates the VPC for them.

Reply:
Then this means users can't add additional security groups if they don't bring a custom VPC. I see that requirement was crossed out in original issue description now which is good. We may need to document this somewhere.